src/RedshiftBase.php

```diff
@@ -100,14 +115,57 @@ private function validateColumns($tableName, $columnsToImport)
         }
     }
 
+    private function insertAllIntoTargetTable($stagingTempTableName, $targetTableName, $primaryKey, $columns, $useTimestamp = true, array $convertEmptyValuesToNull = [])
+    {
+        // create table same as target table
+        $newTargetTableName = $this->createTableFromSourceTable($targetTableName, $primaryKey, $this->schemaName);
+
+        // insert data to new table from staging table
+        $columnsSql = implode(', ', array_map(function ($column) {
+            return $this->quoteIdentifier($column);
+        }, $columns));
+
+        $columnsSelectSql = implode(', ', array_map(function ($column) use ($convertEmptyValuesToNull) {
+            if (in_array($column, $convertEmptyValuesToNull)) {
+                return "CASE {$this->quoteIdentifier($column)}::VARCHAR WHEN '' THEN NULL ELSE {$this->quoteIdentifier($column)} END";
+            }
+            return $this->quoteIdentifier($column);
+        }, $columns));
+
+        $now = $this->getNowFormatted();
+        $newTargetTableNameWithSchema = $this->nameWithSchemaEscaped($newTargetTableName);
+        $stagingTableNameEscaped = $this->tableNameEscaped($stagingTempTableName);
+        if (in_array('_timestamp', $columns) || $useTimestamp === false) {
+            $sql = "INSERT INTO {$newTargetTableNameWithSchema} ($columnsSql) (SELECT $columnsSelectSql FROM $stagingTableNameEscaped)";
+        } else {
+            $sql = "INSERT INTO {$newTargetTableNameWithSchema} ($columnsSql, _timestamp) (SELECT $columnsSelectSql, '{$now}' FROM $stagingTableNameEscaped)";
+        }
+
+        Debugger::timer('copyFromStagingToTarget');
+        $this->query($sql);
+        $this->addTimer('copyFromStagingToTarget', Debugger::timer('copyFromStagingToTarget'));
+
+        // swap tables in transaction
+        $this->connection->beginTransaction();
+
+        $targetTableNameWithSchema = $this->nameWithSchemaEscaped($targetTableName);
+        $this->query(sprintf('DROP TABLE %s', $targetTableNameWithSchema));
+        $this->query(sprintf(
+            "ALTER TABLE %s RENAME TO %s",
+            $newTargetTableNameWithSchema,
+            $this->tableNameEscaped($targetTableName)
+        ));
```

Comment:
mozno je to prkotina ale nebolo by lepsie prehodit drop s rename? tj naprv rename a az potom drop